rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -1170,6 +1218,7 @@ extern "C" rmw_ret_t rmw_destroy_node(rmw_node_t * node)
   rmw_free(const_cast<char *>(node->namespace_));
   rmw_node_free(node);
   delete node_impl;
+  node->context->impl->fini();
```

Comment:
This looks like a use-after-free of `node` to me

Reply:
Oh, that's a good one. Fixed in https://github.com/ros2/rmw_cyclonedds/pull/176/commits/34c9a437c9123f89ac6b7df609b4efc7c37f141a